larq/optimizers_test.py

```diff
@@ -2,49 +2,130 @@
 import numpy as np
 import tensorflow as tf
 import larq as lq
+import distutils.version
 
-if int(tf.__version__[0]) == 2:
-    pytest.skip("This wrapper is not supported by TF 2", allow_module_level=True)
+from larq import testing_utils as lq_testing_utils
+from tensorflow import keras
+from tensorflow.python.keras import testing_utils
+
+
+class AssertLRCallback(keras.callbacks.Callback):
+    def __init__(self, schedule):
+        self.schedule = schedule
+        super().__init__()
+
+    def on_epoch_end(self, epoch, logs=None):
+        learning_rate = keras.backend.get_value(self.model.optimizer.lr)
+        np.testing.assert_allclose(learning_rate, self.schedule(epoch))
 
 
 def assert_weights(weights, expected):
     for w, e in zip(weights, expected):
         np.testing.assert_allclose(np.squeeze(w), e)
 
 
-def test_xavier_scaling():
-    dense = lq.layers.QuantDense(
-        1, kernel_quantizer="ste_sign", kernel_initializer="zeros", input_shape=(1,)
+def _test_optimizer(optimizer, target=0.75):
+    np.random.seed(1337)
+    (x_train, y_train), _ = testing_utils.get_test_data(
+        train_samples=1000, test_samples=0, input_shape=(10,), num_classes=2
     )
-    model = tf.keras.models.Sequential([dense])
+    y_train = keras.utils.to_categorical(y_train)
+
+    model = lq_testing_utils.get_small_bnn_model(x_train.shape[1], 20, y_train.shape[1])
+    model.compile(loss="categorical_crossentropy", optimizer=optimizer, metrics=["acc"])
+    history = model.fit(x_train, y_train, epochs=2, batch_size=16, verbose=0)
+
+    for layer in model.layers:
+        if "quant" in layer.name:
+            assert np.all(np.isin(layer.get_weights(), [-1, 1]))
```

Comment:
This fails for optimizers that rely on latent weights (and rely on `kernel_quantizer="some_quantizer"` to obtain a binarized forward pass). In the future we should probably add a flag for whether to run this test.